x-pack/plugins/security_solution/public/timelines/components/timeline/unified_components/index.test.tsx

```diff
@@ -0,0 +1,629 @@
+}));
+
+jest.mock('../../../containers/details');
+
+jest.mock('../../fields_browser', () => ({
+  useFieldBrowserOptions: jest.fn(),
+}));
+
+jest.mock('../body/events', () => ({
+  Events: () => <></>,
+}));
+
+jest.mock('../../../../common/containers/sourcerer');
+jest.mock('../../../../common/containers/sourcerer/use_signal_helpers', () => ({
+  useSignalHelpers: () => ({ signalIndexNeedsInit: false }),
+}));
+
+jest.mock('../../../../common/lib/kuery');
+
+jest.mock('../../../../common/hooks/use_experimental_features');
+
+const useIsExperimentalFeatureEnabledMock = jest.fn((feature: keyof ExperimentalFeatures) => {
+  if (feature === 'unifiedComponentsInTimelineEnabled') {
+    return true;
+  }
+  return allowedExperimentalValues[feature];
+});
+
+jest.mock('../../../../common/lib/kibana');
+
+// unified-field-list is is reporiting multiple analytics events
+jest.mock(`@kbn/analytics-client`);
+
+const columnsToDisplay = [
+  ...defaultUdtHeaders,
+  {
+    columnHeaderType: 'not-filtered' as ColumnHeaderType,
+    id: 'event.severity',
+  },
+];
+
+const TestComponent = (props: Partial<ComponentProps<typeof UnifiedTimeline>>) => {
+  const testComponentDefaultProps: ComponentProps<typeof QueryTabContent> = {
+    columns: getColumnHeaders(columnsToDisplay, mockSourcererScope.browserFields),
+    activeTab: TimelineTabs.query,
+    rowRenderers: [],
+    timelineId: TimelineId.test,
+    itemsPerPage: 10,
+    itemsPerPageOptions: [],
+    sort: [
+      {
+        columnId: '@timestamp',
+        columnType: 'date',
+        esTypes: ['date'],
+        sortDirection: 'desc',
+      },
+    ],
+    events: structuredClone(mockTimelineData),
+    refetch: jest.fn(),
+    totalCount: mockTimelineData.length,
+    onEventClosed: jest.fn(),
+    expandedDetail: {},
+    showExpandedDetails: false,
+    onChangePage: jest.fn(),
+    dataLoadingState: DataLoadingState.loaded,
+    updatedAt: Date.now(),
+    isTextBasedQuery: false,
+  };
+
+  const dispatch = useDispatch();
+
+  // populating timeline so that it is not blank
+  useEffect(() => {
+    dispatch(
+      timelineActions.applyKqlFilterQuery({
+        id: TimelineId.test,
+        filterQuery: {
+          kuery: {
+            kind: 'kuery',
+            expression: '*',
+          },
+          serializedQuery: '*',
+        },
+      })
+    );
+  }, [dispatch]);
+
+  return <UnifiedTimeline {...testComponentDefaultProps} {...props} />;
+};
+
+const customStore = createMockStore();
+
+const TestProviderWrapperWithCustomStore: React.FC = ({ children }) => {
+  return <TestProviders store={customStore}>{children}</TestProviders>;
+};
+
+const renderTestComponents = (props?: Partial<ComponentProps<typeof TestComponent>>) => {
+  return render(<TestComponent {...props} />, {
+    wrapper: TestProviderWrapperWithCustomStore,
+  });
+};
+
+const loadPageMock = jest.fn();
+
+const useTimelineEventsMock = jest.fn(() => [
+  false,
+  {
+    events: cloneDeep(mockTimelineData),
```

Comment:
you can use the `mockedTimelineData` above

Reply:
Done.